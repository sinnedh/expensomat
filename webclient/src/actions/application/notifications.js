export const resetNotification = () => {
  return {
    type: 'APPLICATION:NOTIFICATION_RESET',
  }
}

export const setErrorNotification = (message) => {
  return {
    type: 'APPLICATION:NOTIFICATION_SET',
    message,
    notificationType: 'error',
  }
}

export const setWarningNotification = (message) => {
  return {
    type: 'APPLICATION:NOTIFICATION_SET',
    message,
    notificationType: 'warning',
  }
}

export const setInfoNotification = (message) => {
  return {
    type: 'APPLICATION:NOTIFICATION_SET',
    message,
    notificationType: 'info',
  }
}

export const setSuccessNotification = (message) => {
  return {
    type: 'APPLICATION:NOTIFICATION_SET',
    message,
    notificationType: 'success',
  }
}
