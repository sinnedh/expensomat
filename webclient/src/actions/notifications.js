export const resetNotification = () => {
  return {
    type: 'NOTIFICATION_RESET',
  }
}

export const setErrorNotification = (message) => {
  return {
    type: 'NOTIFICATION_SET_ERROR',
    message
  }
}

export const setWarningNotification = (message) => {
  return {
    type: 'NOTIFICATION_SET_WARNING',
    message
  }
}

export const setInfoNotification = (message) => {
  return {
    type: 'NOTIFICATION_SET_INFO',
    message
  }
}

export const setSuccessNotification = (message) => {
  return {
    type: 'NOTIFICATION_SET_SUCCESS',
    message
  }
}
