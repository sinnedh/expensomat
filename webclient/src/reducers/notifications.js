const initialState = {
  message: '',
  type: null, // can be 'error', 'warning', 'info', 'success'
}

const notifications = (state = initialState, action) => {
  switch (action.type) {
    case 'NOTIFICATION_RESET':
      return initialState;

    case 'NOTIFICATION_SET_ERROR':
      return {
        message: action.message,
        type: 'error',
      }

    case 'NOTIFICATION_SET_WARNING':
      return {
        message: action.message,
        type: 'warning',
      }

    case 'NOTIFICATION_SET_INFO':
      return {
        message: action.message,
        type: 'info',
      }

    case 'NOTIFICATION_SET_SUCCESS':
      return {
        message: action.message,
        type: 'success',
      }

    default:
      return state;
  }
}

export default notifications;
