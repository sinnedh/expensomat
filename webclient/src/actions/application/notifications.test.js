import * as actions from './notifications'

describe('APPLICATION', () => {
  it('handles resetNotification', () => {
    const action = actions.resetNotification()
    expect(action).toMatchSnapshot()
  })

  it('handles setErrorNotification', () => {
    const action = actions.setErrorNotification('This is an error.')
    expect(action).toMatchSnapshot()
  })

  it('handles setWarningNotification', () => {
    const action = actions.setWarningNotification('This is a warning.')
    expect(action).toMatchSnapshot()
  })

  it('handles setInfoNotification', () => {
    const action = actions.setInfoNotification('This is an info.')
    expect(action).toMatchSnapshot()
  })

  it('handles setSuccessNotification', () => {
    const action = actions.setSuccessNotification('This is a success.')
    expect(action).toMatchSnapshot()
  })
})
