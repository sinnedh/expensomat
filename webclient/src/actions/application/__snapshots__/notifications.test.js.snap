// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APPLICATION handles resetNotification 1`] = `
Object {
  "type": "APPLICATION:NOTIFICATION_RESET",
}
`;

exports[`APPLICATION handles setErrorNotification 1`] = `
Object {
  "notificationText": "This is an error.",
  "notificationType": "error",
  "type": "APPLICATION:NOTIFICATION_SET",
}
`;

exports[`APPLICATION handles setInfoNotification 1`] = `
Object {
  "notificationText": "This is an info.",
  "notificationType": "info",
  "type": "APPLICATION:NOTIFICATION_SET",
}
`;

exports[`APPLICATION handles setSuccessNotification 1`] = `
Object {
  "notificationText": "This is a success.",
  "notificationType": "success",
  "type": "APPLICATION:NOTIFICATION_SET",
}
`;

exports[`APPLICATION handles setWarningNotification 1`] = `
Object {
  "notificationText": "This is a warning.",
  "notificationType": "warning",
  "type": "APPLICATION:NOTIFICATION_SET",
}
`;
